jerry-ext/args/arg-transform-functions.c

```diff
@@ -0,0 +1,249 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerryscript-ext/args.h"
+#include "jerryscript.h"
+
+/* Common function for optional argument */
+
+static jerry_value_t
+jerryx_arg_transform_optional(jerryx_arg_iterator_t *iterator_p,
+                              jerryx_arg_transform_func_t func)
+{
+  jerry_value_t peeked_js_arg = jerryx_arg_iterator_peek (iterator_p);
+
+  if (jerry_value_is_undefined (peeked_js_arg))
+  {
+    jerryx_arg_iterator_pop (iterator_p);
+    return jerry_create_undefined ();
+  }
+
+  return func(iterator_p);
+}
+
+
+/* Number transform function. */
+
+static jerry_value_t
+jerryx_arg_transform_number_strict_impl (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  if (!jerry_value_is_number (js_arg))
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE,
+                               (jerry_char_t *) "It is not a number.");
+  }
+
+  double *dest = jerryx_arg_iterator_get_dest (iterator_p);
+  *dest = jerry_get_number_value (js_arg);
+
+  return jerry_create_undefined ();
+}
+
+static jerry_value_t
+jerryx_arg_transform_number_impl (jerryx_arg_iterator_t *iterator_p)
+{
+  jerry_value_t js_arg = jerryx_arg_iterator_pop (iterator_p);
+
+  jerry_value_t to_number = jerry_value_to_number (js_arg);
+
+  if (jerry_value_has_error_flag (to_number))
+  {
+    jerry_release_value (to_number);
```

Comment:
For example, the js_arg is a jerry_error. https://github.com/jerryscript-project/jerryscript/blob/master/jerry-core/jerry.c#L731